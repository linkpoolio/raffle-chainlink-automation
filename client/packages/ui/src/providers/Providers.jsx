import React from 'react'
import { Router } from 'react-router-dom'
import { Web3WalletProvider } from './'

export const Providers = ({ children, history }) => (
  <Router history={history}>
    <Web3WalletProvider>{children}</Web3WalletProvider>
  </Router>
)
