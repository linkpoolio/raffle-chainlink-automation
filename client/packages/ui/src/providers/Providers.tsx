import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import {
  ConnectKitProvider,
  ConnectKitButton,
  getDefaultClient
} from 'connectkit'
import { WagmiConfig, createClient } from 'wagmi'

import { client } from '@ui/config/client'
import { Web3WalletProvider } from './'

const history = createBrowserHistory()

export const Providers = ({ children }) => (
  <Router history={history}>
    <WagmiConfig client={client}>
      <ConnectKitProvider>
        <Web3WalletProvider>{children}</Web3WalletProvider>
      </ConnectKitProvider>
    </WagmiConfig>
  </Router>
)