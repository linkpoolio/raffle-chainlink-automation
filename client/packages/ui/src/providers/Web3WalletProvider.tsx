import React from 'react'
import { ethers } from 'ethers'
import { WagmiConfig, createClient } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { mainnet } from 'wagmi/chains'

const defaultChains = [mainnet]

const getProvider = () =>
  new ethers.providers.JsonRpcProvider('https://main-light.eth.linkpool.io', 1)

const connectors = () => {
  return [
    new InjectedConnector({
      chains: defaultChains,
      options: { shimDisconnect: true }
    })
  ]
}

const client = createClient({
  autoConnect: true,
  connectors,
  provider: getProvider
})

export const Web3WalletProvider = ({ children }) => (
  <WagmiConfig client={client}>{children}</WagmiConfig>
)
