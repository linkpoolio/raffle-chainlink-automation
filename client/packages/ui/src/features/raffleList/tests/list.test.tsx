import React from 'react'
import renderer from 'react-test-renderer'

import { Providers } from '@ui/providers'
import { RaffleList } from '../'

const getComponent = () => {
  const component = (
    <Providers>
      <RaffleList />
    </Providers>
  )

  return component
}

describe('RaffleList', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
