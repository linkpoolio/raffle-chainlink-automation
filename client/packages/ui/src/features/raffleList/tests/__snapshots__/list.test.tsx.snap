// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaffleList matches snapshot 1`] = `
[
  <div
    className="chakra-container css-1xetpgu"
  >
    <div
      className="css-mzt97k"
    >
      <div
        className="chakra-stack css-1y5q8ss"
      >
        <div
          className="chakra-select__wrapper css-4j45yy"
        >
          <select
            className="chakra-select css-1psspw3"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value={0}
            >
              Staged
            </option>
            <option
              value={1}
            >
              Live
            </option>
            <option
              value={2}
            >
              Finished
            </option>
          </select>
          <div
            className="chakra-select__icon-wrapper css-iohxn1"
          >
            <svg
              aria-hidden={true}
              className="chakra-select__icon"
              focusable={false}
              role="presentation"
              style={
                {
                  "color": "currentColor",
                  "height": "1em",
                  "width": "1em",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1db7fom"
    />
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
